packages/cactus-plugin-keychain-google-sm/src/main/typescript/plugin-keychain-google-sm.ts

```diff
@@ -0,0 +1,182 @@
+  backend?: SecretManagerServiceClient;
+}
+
+export class PluginKeychainGoogleSm
+  implements ICactusPlugin, IPluginWebService, IPluginKeychain {
+  public static readonly CLASS_NAME = "PluginKeychainGoogleSm";
+  private readonly log: Logger;
+  private readonly instanceId: string;
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private googleSmClient: SecretManagerServiceClient;
+
+  public get className() {
+    return PluginKeychainGoogleSm.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: IPluginKeychainGoogleSmOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.keychainId, `${fnTag} arg options.keychainId`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    Checks.nonBlankString(opts.keychainId, `${fnTag} options.keychainId`);
+
+    if (opts.backend) {
+      this.googleSmClient = opts.backend;
+    } else {
+      this.googleSmClient = new SecretManagerServiceClient();
+    }
+    this.instanceId = this.opts.instanceId;
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  async registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+    const endpoints: IWebServiceEndpoint[] = [];
+
+    // TODO: Writing the getExpressRequestHandler() method for
+    // GetKeychainEntryEndpointV1 and SetKeychainEntryEndpointV1
+
+    // {
+    //   const ep = new GetKeychainEntryEndpointV1({
+    //     logLevel: this.opts.logLevel,
+    //   });
+    //   ep.registerExpress(expressApp);
+    //   endpoints.push(ep);
+    // }
+    // {
+    //   const ep = new SetKeychainEntryEndpointV1({
+    //     logLevel: this.opts.logLevel,
+    //   });
+    //   ep.registerExpress(expressApp);
+    //   endpoints.push(ep);
+    // }
+    this.endpoints = endpoints;
+
+    return endpoints;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  public async shutdown(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getKeychainId(): string {
+    return this.opts.keychainId;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-keychain-vault`;
+  }
+
+  async rotateEncryptionKeys(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getEncryptionAlgorithm(): string {
+    return null as any;
```

Comment:
@jagpreetsinghsasan Is there any documentation about the encryption algorithm used by GCP SM?

Reply:
https://cloud.google.com/secret-manager/docs/encryption

I think it is **AES-256**